src/main/scala/exercises/06_Collections.scala

```diff
@@ -8,161 +8,166 @@ import scala.collection.mutable.ListBuffer
    *  - custom made
-   *  - ... etc 
+   *  - ... etc
    */
   def concatLists[T](l1: List[T], l2: List[T]): List[T] = {
-    error("fix me")
+    l1 ::: l2
   }
 
   /**
    * Sort a list on the natural ordering, so sortList(3,1,2) = List(1,2,3).
-   * 
+   *
    * Hint: this can be achieved in multiple ways:
    * - built in
    * - via a foldLeft method (a bit complex, but fun)
-   * - ... whichever way you like 
-   * 
+   * - ... whichever way you like
+   *
    */
   def sortList[T <% Ordered[T]](list: List[T]): List[T] = {
-    error("fix me")
+    list.foldLeft(List[T]()) { (r, c) =>
+      val (front, back) = r.span(_ < c)
+      front ::: c :: back
+    }
   }
 
   /**
-   * Check whether a given element in a list exists, 
+   * Check whether a given element in a list exists,
    * i.e. elementExists(List("a", "b", "c"), "b") = true
-   * 
-   * Again, easy to implement using built-in functionality, but also possible 
+   *
+   * Again, easy to implement using built-in functionality, but also possible
    * to implement in your own free-style way.
    */
   def elementExists[T](l: List[T], e: T): Boolean = {
-    error("fix me")
+    l.foldLeft(false)(_ || _ == e)
   }
 
   /**
-   * Get all odd elements in the list, i.e. 
+   * Get all odd elements in the list, i.e.
    * oddElements(List(1,2,3,4,5)) = List(1,3,5)
-   * 
+   *
    * As always, use either build-in functions, or roll your own way via a
    * pattern match or some other method.
    */
   def oddElements(iList: List[Int]): List[Int] = {
-    error("fix me")
+    iList.filter(_ % 2 == 0)
   }
 
   /**
-   * Returns a list of lists, containing all final segments of the argument 
+   * Returns a list of lists, containing all final segments of the argument
    * list, longest first.
-   * 
+   *
    * For example: tails(List(1,2,3,4)) = List(List(1,2,3,4), List(2,3,4), List(3,4), List(4), List())
-   * 
-   * Inspired by Haskell's tails function: 
+   *
+   * Inspired by Haskell's tails function:
    * 	http://www.zvon.org/other/haskell/Outputlist/tails_f.html
-   * 
-   * Implement it whatever way suites you best. Hint: it can be done in a 
-   * neat way using recursion. 
+   *
+   * Implement it whatever way suites you best. Hint: it can be done in a
+   * neat way using recursion.
    */
   def tails[T](l: List[T]): List[List[T]] = {
-    error("fix me")
+    if (l.length == 0) List(List()) else l::tails(l.tail)
   }
-  
+
   /**
    * Find the maximum element in a list, e.g. maxElementInList(List(1,9,3,5)) == 9
-   * 
+   *
    * As usual, various ways exist: pattern matching, folding, ...
    */
   def maxElementInList(l: List[Int]): Int = {
-    error("fix me")
+    l.foldLeft(l.head){(r, c) =>
+      if (r > c) r else c
+    }
   }
 
   /**
    * Calculate the sum of the equally position elements
    * of the two list
    */
   def sumOfTwo(l1: List[Int], l2: List[Int]): List[Int] = {
-    error("fix me")
+    l1.zipAll(l2, 0, 0).map {case (a, b) => a + b}
```

Comment:
Clever way of using the zipAll method, I like how you've been using a variety of different built-in methods!